frontend/packages/dev-console/src/utils/truncate-utils.ts

```diff
@@ -0,0 +1,49 @@
+export type TruncateOptions = {
+  length?: number; // Length to truncate text to
+  middle?: boolean; // Flag if the text should be truncated in the middle (false for end)
+  omission?: string; // Truncation text used to denote the truncation (ellipsis)
+  minTruncateChars?: number; // Minimum number of characters to truncate
+};
+
+const DEFAULT_OPTIONS: TruncateOptions = {
+  length: 20,
+  middle: true,
+  omission: '\u2026', // ellipsis character
+  minTruncateChars: 3,
+};
+
+// Truncates a string down to `maxLength` characters when the length
+// is greater than the `maxLength` + `minTruncateChars` values.
+// By default the middle is truncated, set the options.middle to false to truncate at the end.
+// Truncated text is replaced with the provided omission option (ellipsis character by default);
+export const truncateText = (text: string, options: TruncateOptions = {}): string => {
```

Comment:
FYI, we do have an existing `truncateMiddle` helper: https://github.com/spadgett/console/blob/master/frontend/public/components/utils/truncate-middle.ts

Reply:
Yeah, this util is a bit different. I tried to use `truncateMiddle` but it fails as this is done as part of initialization of the plugin.

Reply:
Oh ya there's that util...
@jeff-phillips-18 can you explain further what the issue is with using that util?
Can we enhance the existing util with what we need. There shouldn't be an issue re-using the module.

Reply:
Was having some issues importing it from `@console/internal/components/utils`, but it works fine if I import it from `@console/internal/components/utils/truncate-middle`.

I've enhanced the existing util and imported as above.